lib/broadway/producer.ex

```diff
@@ -27,27 +27,35 @@ defmodule Broadway.Producer do
   @optional_callbacks prepare_for_start: 2, prepare_for_draining: 1
 
   @spec start_link(term, GenServer.options()) :: GenServer.on_start()
-  def start_link(args, opts \\ []) do
-    GenStage.start_link(__MODULE__, args, opts)
+  def start_link(args, index, opts \\ []) do
+    GenStage.start_link(__MODULE__, {args, index}, opts)
   end
 
-  @spec push_messages(GenServer.server, [Message.t()]) :: :ok
+  @spec push_messages(GenServer.server(), [Message.t()]) :: :ok
   def push_messages(producer, messages) do
     GenStage.call(producer, {__MODULE__, :push_messages, messages})
   end
 
-  @spec drain(GenServer.server) :: :ok
+  @spec drain(GenServer.server()) :: :ok
```

Comment:
Sorry, these changes got in because of the formatter. I didn't remove them since I think it's a good thing™ to have the code formatted 😄